shttps/ChunkReader.cpp

```diff
@@ -49,10 +49,16 @@ using namespace std;
 
 namespace shttps {
 
-    static size_t read_chunk(istream &ins, char **buf, size_t offs = 0)
+    ChunkReader::ChunkReader(std::istream *ins_p, size_t post_maxsize_p) : ins(ins_p), post_maxsize(post_maxsize_p) {
+        chunk_size = 0;
+        chunk_pos = 0;
+    };
+    //=========================================================================
+
+    size_t ChunkReader::read_chunk(istream &ins, char **buf, size_t offs)
     {
         string line;
-        (void) safeGetline(ins, line); // read empty line
+        (void) safeGetline(ins, line); // read chunk size
         if (ins.fail() || ins.eof()) {
             throw -1;
```

Comment:
No – this is finally not an error. It just means that the client has cancelled the connection. But I could make a typedef to have a better name for this case...